ibis/common/tests/test_patterns.py

```diff
@@ -495,6 +554,60 @@ def test_matching():
     assert match(InstanceOf(object) & InstanceOf(float), 3.14) == 3.14
 
 
+def test_replace_passes_matched_value_as_underscore():
+    class MyInt:
+        def __init__(self, value):
+            self.value = value
+
+        def __eq__(self, other):
+            return self.value == other.value
+
+    p = InstanceOf(int) >> Call(MyInt, value=_)
```

Comment:
What is the difference between `Call` and `Object`?

Reply:
Call is `Creator` (awful naming I know) while `Object` is a pattern. The difference between the two is the former only receives the `context` with the captured values to *construct* a new object which is substituted by the `Replace` pattern. The latter does matching on objects considering both the objects' type and arguments.

Reply:
I guess the question is whether the nouns here (`Call`, `Object`, etc.) are referring to the identities of the match candidate ("this is a `Call`") or whether the nouns are referring to properties of the match candidate ("this was `CalledWith`").

It seems like you've gone with the first option, which is fine!